types/rdfjs__data-model/index.d.ts

```diff
@@ -0,0 +1,22 @@
+// Type definitions for @rdfjs/data-model 2.0
+// Project: https://github.com/rdfjs-base/data-model
+// Definitions by: Jesse Wright <https://github.com/jeswr>
+//                 Ruben Taelman <https://github.com/rubensworks>
+//                 Andy Davison  <https://github.com/andydavison>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.9
+
+import * as RDF from "@rdfjs/types";
+
+interface Factory<OutQuad extends RDF.BaseQuad = RDF.Quad, InQuad extends RDF.BaseQuad = OutQuad>
+  // Required is needed here because variable is optional in the spec
+  extends Required<RDF.DataFactory<OutQuad, InQuad>> {
```

Comment:
Why the generic arguments? The factory itself is not generic but a concrete implementation which always returns `Quad`. Might use `BaseQuad` as "in" arg for compatibility. I think it would suffice like

```suggestion
// Required is needed here because variable is optional in the spec
interface Factory extends Required<RDF.DataFactory<RDF.BaseQuad, RDF.Quad>> {
```

Reply:
I agree that this does not need to be generic, but why have the `InQuad` as an `RDF.Quad` rather than an `RDF.BaseQuad`?

Reply:
Sorry, my mistake. Obviously I got the type args the other way round

Reply:
I'm actually thinking they should both be `RDF.BaseQuad` - in [their implementation of quad](https://github.com/rdfjs-base/data-model/blob/master/lib/Quad.js) there is nothing to enforce that it returns a quad, and this can return `RDF.BaseQuad`s if you want it to.

As far as I understand it, this is in contradiction to the earlier comment
> The factory itself is not generic but a concrete implementation which always returns `Quad`

Reply:
Yes, I realise that now. Still, I suppose `RDF.Quad` is a sane default

Reply:
Resolving per https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57899#discussion_r776648895